### PR TITLE
Gsplat like depth loss + depth map viewer

### PR DESCRIPTION
**Pull request by Deepthought73**
I added an L1 loss on the inverse depth, just like gsplat implements the depth loss. At the moment, the depth maps are loaded as TIFF float32 images from a folder named depth, only for COLMAP datasets. Since COLMAP is scale ambiguous, I added an argument that uses the depth maps and the projected landmarks to estimate the correct scale of the COLMAP dataset.

Here is a video with a comparison [https://www.youtube.com/watch?v=Si_vElXaDjo](https://www.youtube.com/watch?v=Si_vElXaDjo) between with and without depth loss.

I also added a depth map viewer with the turbo color map:
image

Please tell me your opinion on my changes. 

Thanks,
Kilian Northoff

### CODE REVIEW

**Comment by debsahu** [comment]
We have been using this PR's depth loss on metric depth maps in a fork. One small suggestion. The rendered depth divides by accumulated alpha in `crates/brush-train/src/train.rs` (the `accumulated_depth / alpha.clamp_min(1e-10)` line in the depth loss block), and the denominator stays differentiable. Depth error can then flow into opacity. Gaussians can lower depth loss by changing transparency instead of moving. LichtFeld Studio detaches the blending weights for this reason, and DN-Splatter does the same. Detaching alpha in the denominator is a one-line change. Happy to open a PR if that helps.

File: crates/brush-train/src/train.rs [hunk 1]
```diff
@@ -253,6 +266,16 @@ impl SplatTrainer {
                     ) * self.config.lpips_loss_weight;
             }
 
+            // Depth Disparity L1 loss on rendered expected depth
+            if use_depth && let Some(depth_data) = &batch.depth {
+                let gt_depth: Tensor<2> = Tensor::from_data(depth_data.clone(), &device);
+                let accumulated_depth = pred_image.clone().slice(s![.., .., 4..5]);
+                let alpha = pred_image.clone().slice(s![.., .., 3..4]);
```
Comment (debsahu) [inline crates/brush-train/src/train.rs]: Here is the one-line version of the detach from my earlier comment. This keeps depth supervision on positions and stops depth error from flowing into opacity.

```suggestion
                let alpha = pred_image.clone().slice(s![.., .., 3..4]).detach();
```

Comment (debsahu) [reply crates/brush-train/src/train.rs]: One correction after a closer look. This detach closes one of two coupling routes. The rasterizer backward also feeds the depth channel gradient into the alpha term through the dot accumulator, so opacity still receives some depth gradient even with the detach. Full decoupling in the LFS sense needs that depth term dropped from the backward too. The detach here stays worthwhile on its own as the cheaper half.

Comment (debsahu) [reply crates/brush-train/src/train.rs]: Opened the complete change as a PR against this branch here: https://github.com/Deepthought73/brush/pull/1
